src/mnemonics/electrum-words.cpp

```diff
@@ -339,6 +339,17 @@ namespace crypto
       return true;
     }
 
+    bool words_to_bytes(std::string words, legacy16B_secret_key& dst, std::string &language_name)
+    {
+      std::string s;
+      if (!words_to_bytes(words, s, sizeof(dst), true, language_name))
+        return false;
+      if (s.size() != sizeof(dst))
+        return false;
+      dst = *(const legacy16B_secret_key*)s.data();
```

Comment:
Make this a `memcpy`, this is an aliasing violation.

Reply:
Updated with 2141411; kept it as a didn't squash this commit because the aliasing violation appears to exist in the existing `crypto::secret_key` variation of `words_to_bytes` (which I was following) so I corrected both as a visible change.